### PR TITLE
docs(theme): Include default css theme to the examples

### LINKED ISSUE #245: documentation(hierarchicalWidget): give UX guidance on the number of levels
When documenting the hierarchical widget, we should advise people to not go too far in terms of hierarchical levels. Because this will lead to UX fails, so two level deep should be sufficient for most instant search experiences.

TODO:
- [ ] on the website documentation, advise not to go too far

### LINKED ISSUE #507: documentation: rendering of hierarchicalMenu is not correctly indented
Seems like the default theme is not applied, you cannot see that this menu is actually hierarchical.

2015-11-09_10h46m22

### PR DESCRIPTION
**Pull request by pixelastic**
Fixes #507

Same as for the js file, it uses the development version when in dev, or the one form jsdelivr in prod.

I've removed the hierarchical indentation that is now part of default. I've moved specific example styling of the search box to the `default-theme.css`, along with other styling.

We might need to rename that file in the future, and move default styling away from this file.

### CODE REVIEW

**Comment by pixelastic** [comment]
Updated, now ready for review.

File: docs/_layouts/default.html [hunk 1]
```diff
@@ -9,7 +9,12 @@
     <meta name="description" content="{{ site.description }}">
 
     <link rel="stylesheet" href="{{ "/css/main.css" | prepend: site.baseurl }}">
-    <link rel="stylesheet" href="{{ "/css/default-theme.css" | prepend: site.baseurl}}" />
+    {% if site.env == 'development' %}
+      <link rel="stylesheet" href="http://localhost:8080/instantsearch.css" />
```
Comment (vvo) [inline docs/_layouts/default.html]: The only difference with the js loading is that in development for example @Shipow that is not running `npm run dev` will not be able to see the default css file.

I think this is part of a bigger issue where the development mode of the docs is dependant of those files and thos files might not exists.

I dunno what we can do here, any idea?

Comment (pixelastic) [inline docs/_layouts/default.html]: As discussed, let's do an `npm run docs` that starts the correct npm, guard and jekyll commands so this is not an issue.

Comment (vvo) [inline docs/_layouts/default.html]: Done #515

**Pull request merged**